Common/Data/Custom/SmartInsider/SmartInsiderEvent.cs

```diff
@@ -0,0 +1,276 @@
+         * public string UserField63 { get; protected set; }
+         * public string UserField64 { get; protected set; }
+         * public string UserField65 { get; protected set; }
+         * public string SystemField66 { get; protected set; }
+         * public string SystemField67 { get; protected set; }
+         * public string SystemField68 { get; protected set; }
+         * public string SystemField69 { get; protected set; }
+         * public string SystemField70 { get; protected set; }
+         * public decimal? UserField71 { get; protected set; }
+         * public decimal? UserField72 { get; protected set; }
+         * public decimal? UserField73 { get; protected set; }
+         * public decimal? UserField74 { get; protected set; }
+         * public decimal? UserField75 { get; protected set; }
+         * public decimal? SystemField76 { get; protected set; }
+         * public decimal? SystemField77 { get; protected set; }
+         * public decimal? SystemField78 { get; protected set; }
+         * public decimal? SystemField79 { get; protected set; }
+         * public decimal? SystemField80 { get; protected set; }
+         * public int? Field81 { get; protected set; }
+         * public int? Field82 { get; protected set; }
+         * public int? Field83 { get; protected set; }
+         * public int? Field84 { get; protected set; }
+         * public int? Field85 { get; protected set; }
+         * public int? Field86 { get; protected set; }
+         * public int? Field87 { get; protected set; }
+         * public int? Field88 { get; protected set; }
+         * public int? Field89 { get; protected set; }
+         * public int? Field90 { get; protected set; }
+         * public DateTime? Field91 { get; protected set; }
+         * public DateTime? Field92 { get; protected set; }
+         * public DateTime? Field93 { get; protected set; }
+         * public DateTime? Field94 { get; protected set; }
+         * public DateTime? Field95 { get; protected set; }
+         * public DateTime? Field96 { get; protected set; }
+         * public DateTime? Field97 { get; protected set; }
+         * public DateTime? Field98 { get; protected set; }
+         * public DateTime? Field99 { get; protected set; }
+         * public DateTime? Field100 { get; protected set; }
+         */
+
+        #endregion
+
+        /// <summary>
+        /// Empty constructor required for cloning
+        /// </summary>
+        public SmartInsiderEvent()
+        {
+        }
+
+        /// <summary>
+        /// Parses a line of CSV (tab delimited) from Smart Insider data
+        /// </summary>
+        /// <param name="csvLine">Tab delimited line of data</param>
+        public SmartInsiderEvent(string csvLine)
+        {
+            var csv = csvLine.Split('\t');
+            var count = csv.Length;
+
+            TransactionID = csv[0];
+            BuybackType = string.IsNullOrWhiteSpace(csv[1]) ? null : csv[1];
```

Comment:
Thanks for making all of this culture-aware -- soon we'll have the `StringExtensions` class merged and should use those methods moving forward. This should help to keep these parsing sections more legible while removing noise from specifying the `CultureInfo` explicitly. Also, the `IfNotNullOrEmpty` extension can further help compact the code by removing the need for the ternary expression:
``` csharp
LastUpdate = csv[2].ParseDateTimeExactInvariant("yyyyMMdd");
LastIDsUpdate = csv[3].IfNotNullOrEmpty<DateTime?>(s => s.ParseDateTimeExactInvariant("yyyyMMdd"));
```

Reply:
As far as I'm aware, I don't think the `IsNotNullOrEmpty` extension is present in the codebase. A quick search of it doesn't bring anything up, but if it existed, I'd definitely use it